llvm/lib/IR/Type.cpp

```diff
@@ -845,6 +852,19 @@ static TargetTypeInfo getTargetTypeInfo(const TargetExtType *Ty) {
     return TargetTypeInfo(ScalableVectorType::get(Type::getInt1Ty(C), 16),
                           TargetExtType::HasZeroInit);
 
+  // RISC-V vector tuple type. The layout is represented as the type that needs
+  // the same number of vector registers(VREGS) as this tuple type, represented
+  // as <vscale x (VREGS * 8) x i8>.
+  if (Name == "riscv.vector.tuple") {
+    unsigned TotalNumElts =
+        std::max(cast<ScalableVectorType>(Ty->getTypeParameter(0))
+                     ->getMinNumElements(),
+                 8U) *
```

Comment:
Where does the 8 come from?

Reply:
I was thinking that fractional `LMUL` needs to align to `LMUL=1` which is `<vscale x 8 x i8>`, but seems it doesn't need to, right?

Reply:
Then we need to use RVVBitsPerBlock here. We can't hard code 8.

Reply:
Got it!